src/main/java/org/testcontainers/vault/VaultContainer.java

```diff
@@ -0,0 +1,139 @@
+package org.testcontainers.vault;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.testcontainers.shaded.com.github.dockerjava.api.model.Capability.IPC_LOCK;
+
+/**
+ * GenericContainer subclass for Vault specific configuration and features. The main feature is the
+ * withSecretInVault method, where users can specify which secrets to be pre-loaded into Vault for
+ * their specific test scenario.
+ *
+ * Other helpful features include the withVaultPort, and withVaultToken methods for convenience.
+ */
+public class VaultContainer<SELF extends VaultContainer<SELF>> extends GenericContainer<SELF>
+        implements LinkableContainer {
+
+    private boolean preloadSecret = false;
+
+    private boolean vaultPortRequested = false;
+
+    private Map<String, List<String>> secretsMap = new HashMap<>();
+
+    public VaultContainer() {
+        this("vault:0.7.0");
+    }
+
+    public VaultContainer(String dockerImageName) {
+        super(dockerImageName);
+    }
+
+    @Override
+    protected void configure() {
+        setStartupAttempts(3);
+        withCreateContainerCmdModifier(cmd -> cmd.withCapAdd(IPC_LOCK));
+        if(!isVaultPortRequested()){
+            withEnv("VAULT_ADDR", "http://0.0.0.0:8200");
+            setPortBindings(Arrays.asList("8200:8200"));
+        }
+    }
+
+    @Override
+    protected void waitUntilContainerStarted() {
+        getWaitStrategy().waitUntilReady(this);
+
+        //add secret to vault
+        if (isPreloadSecret()) {
+            addSecrets();
+        }
+    }
+
+    private void addSecrets() {
+        secretsMap.forEach((path, secrets) -> {
```

Comment:
I have reworked the method to now run a single exec command even if there are multiple secrets or paths.